web/src/compose_state.ts

```diff
@@ -133,6 +134,27 @@ export function rewire_topic(value: typeof topic): void {
     topic = value;
 }
 
+export function present_channel_message_recipient_topic(topic_name: string | undefined): void {
+    if (typeof topic_name !== "string") {
+        topic_name = "";
+    }
+    // When topics are mandatory, we just call topic() as usual.
+    // Otherwise, we add a class to specially style the placeholder
+    // text to look more like the usual topic input.
+    if (realm.realm_mandatory_topics) {
+        topic(topic_name);
+        return;
+    }
+    const $input = $("input#stream_message_recipient_topic");
+    const is_empty_stream_topic = topic_name === "";
+    if (is_empty_stream_topic) {
+        $input.addClass("empty-topic-display");
+    } else {
+        $input.removeClass("empty-topic-display");
+    }
+    topic(topic_name);
+}
```

Comment:
Is there a way to do this inside `compose_recipient.ts` without a lot of hassle? I think that function is traditionally responsible for the UI, and this one just storing the state, but they're a bit entangled. Not a blocker.

@evykassirer may be able to suggest a good way to do it.